### PR TITLE
[Doc]Update 22.06 documentation[skip ci]

### LINKED ISSUE #5217: [DOC] FAQ should clarify why Spark's Java and R APIs are not tested
The FAQ states that the Java and R APIs are supported but not tested.  This can confuse users into thinking it's really _not_ supported since it's completely untested.  However the reason we do not focus on those APIs is because it would be like testing Spark itself.  The RAPIDS Accelerator hooks into Spark not at the various language APIs but at the Catalyst level after all the various APIs have converged into the DataFrame API.  Clarifying why these are untested can give users more confidence in using the RAPIDS Accelerator with these Spark APIs.

### PR DESCRIPTION
**Pull request by viadea**
Fixes #5217

1. Add download page for 22.06. 
(Some of the features are not ready yet such as Spark 3.3 support, so I will add later once it is merged in 22.06 branch)

2. Address https://github.com/NVIDIA/spark-rapids/issues/5460

3. Address https://github.com/NVIDIA/spark-rapids/issues/5217 

4. Add K8s doc to mention the base CUDA images and its dockerfile.

5. Modify the examples README to point to spark-rapids-examples and spark-rapids-benchmark repos.

6. Swap two steps in Alluxio getting-stated doc because you can not run command `mount` before starting the alluxio cluster. 

7. Some other minor doc update

### CODE REVIEW

File: docs/FAQ.md
Comment (jlowe) [inline docs/FAQ.md:310]: We should have the same text for the Java APIs FAQ just below, as a user wanting to use the Java Spark APIs thought it would be risky to do so the way it was worded.

File: docs/download.md
Comment (jlowe) [inline docs/download.md:46]: Because of this currently bad link, I'd like to see this checked in as late as possible.  Otherwise we end up with every PR in the meantime being flagged for a bad link because it's checked in that way.

File: docs/download.md
Comment (jlowe) [inline docs/download.md:49]: CUDA forward compatibility is not something we're enabling.  That's a feature of CUDA, not this project.  As such I would word this differently, e.g.: "Older CUDA 11.x versions are supported through CUDA forward compatibility."

File: docs/download.md
Comment (jlowe) [inline docs/download.md:51]: Should say "CUDA 11.5 or later is required" here, as CUDA backward compatibility will allow us to run on CUDA versions > 11.5.

File: docs/download.md
Comment (jlowe) [inline docs/download.md:67]: We should more clearly mention here that the methods for specifying jars on the Spark client command-line have changed and ideally link to documentation for more details.  Failing that, we minimally need to state that the RAPIDS Accelerator jar is the only jar that needs to be passed to Spark and the cudf jar should _not_ be specified.

File: docs/get-started/getting-started-kubernetes.md
Comment (jlowe) [inline docs/get-started/getting-started-kubernetes.md:22]:
```suggestion
[GitLab repository](https://gitlab.com/nvidia/container-images/cuda/) which can be used to build
```

File: docs/tuning-guide.md
Comment (jlowe) [inline docs/tuning-guide.md:202]:
```suggestion
We suggest reordering the columns needed by the queries and then rewrite the files to make those
```

File: docs/download.md
Comment (viadea) [reply docs/download.md:46]: Yes we can wait for some time to merge this PR. 
My plan is to merge this PR  before the merge request to main, so that future gh-pages update PR can take it from there.

**Comment by viadea** [comment]
> @viadea could we add a FAQ entry to say the ASYNC allocator is on by default but for CUDA 11.4 and older drivers we will fallback to ARENA.

Added. How about now?

File: docs/FAQ.md
Comment (viadea) [reply docs/FAQ.md:310]: Done.

File: docs/download.md
Comment (viadea) [reply docs/download.md]: Changed.

File: docs/download.md
Comment (viadea) [reply docs/download.md]: Changed.

File: docs/download.md
Comment (viadea) [reply docs/download.md]: Changed. How about now?

File: docs/download.md
Comment (jlowe) [inline docs/download.md:68]:
```suggestion
  The RAPIDS Accelerator jar (rapids-4-spark jar) is the only jar that needs to be passed to Spark 
```

**Review by jlowe — DISMISSED** [review]

File: docs/download.md
Comment (sameerz) [inline docs/download.md:76]: Enabling CSV reads, regular expressions, and floating point operations by default ought to be higher on the list of new features.  spark.sql.mapKeyDedupPolicy=LAST_WIN is probably not that important to highlight.  Rather, we can highlight features like: Improved ANSI support, Supporting for Avro reading of primitive types,

File: docs/tuning-guide.md
Comment (sameerz) [inline docs/tuning-guide.md:204]: Should we add a comment here about using `spark.rapids.sql.format.parquet.reader.footer.type=NATIVE` if there are a large number of columns and the data format is Parquet?

Comment (jlowe) [reply docs/tuning-guide.md:204]: The feature is experimental.  Not sure we're ready to widely advertise it yet, but I'd defer to @revans2 on this.

File: docs/download.md
Comment (viadea) [reply docs/download.md:72]: Refactored the release notes.

BTW: for "Avro reading of primitive types" it was added for 22.04 before.

File: docs/tuning-guide.md
Comment (sameerz) [reply docs/tuning-guide.md:204]: Fair enough, we can add the note about it in the tuning guide after it is no longer experimental.

File: docs/download.md
Comment (sameerz) [reply docs/download.md:72]: Got it, thanks.

**Comment by sameerz** [comment]
Do we need to update other parts of the documentation where we refer to the cudf jar, such as:
* getting-started-on-prem.md, which has download instructions for cudf and scripts which reference the cudf jar location
* generate-init-script.ipynb, which downloads and sets environment variables for the cudf jar
* getting-started-gcp.md , which mentions "The notebook depends on the pre-compiled [Spark RAPIDS SQL plugin](https://mvnrepository.com/artifact/com.nvidia/rapids-4-spark) and [cuDF](https://mvnrepository.com/artifact/ai.rapids/cudf), which are pre-downloaded by the GCP Dataproc [RAPIDS init script](https://github.com/GoogleCloudDataproc/initialization-actions/tree/master/rapids)."
* getting-started-kubernetes.md, which has download instructions for cudf and scripts which reference the cudf jar location
* getting-started-workload-qualification, which has instructions on downloading and using the cudf jar
* spark-profiling-tool.md, which mentions the cudf jar as a depdendancy
* additional-functionality/rapids-shuffle.md, which includes the cudf jar in the classpath
* dev/nvtx_profiling.md might need updating, as it mentions compiling the cudf jar separately

**Comment by viadea** [comment]
> Do we need to update other parts of the documentation where we refer to the cudf jar, such as:
> 
>     * getting-started-on-prem.md, which has download instructions for cudf and scripts which reference the cudf jar location
> 
>     * generate-init-script.ipynb, which downloads and sets environment variables for the cudf jar
> 
>     * getting-started-gcp.md , which mentions "The notebook depends on the pre-compiled [Spark RAPIDS SQL plugin](https://mvnrepository.com/artifact/com.nvidia/rapids-4-spark) and [cuDF](https://mvnrepository.com/artifact/ai.rapids/cudf), which are pre-downloaded by the GCP Dataproc [RAPIDS init script](https://github.com/GoogleCloudDataproc/initialization-actions/tree/master/rapids)."
> 
>     * getting-started-kubernetes.md, which has download instructions for cudf and scripts which reference the cudf jar location
> 
>     * getting-started-workload-qualification, which has instructions on downloading and using the cudf jar
> 
>     * spark-profiling-tool.md, which mentions the cudf jar as a depdendancy
> 
>     * additional-functionality/rapids-shuffle.md, which includes the cudf jar in the classpath
> 
>     * dev/nvtx_profiling.md might need updating, as it mentions compiling the cudf jar separately

@sameerz  

I think most of the above were already handled by previous PRs in 22.06 branch. 
But I did some update/fix in the latest commit:
1. getting-started-gcp.md -- Fixed
2. getting-started-databricks.md -- removed "with RAPIDS and cuDF" words.
3. additional-functionality/rapids-shuffle.md -- Fixed.
4. dev/nvtx_profiling.md -- Since we no longer need to build the cuDF jar ourselves so i just removed the whole step 1.


Regarding "spark-profiling-tool.md", my thought is our profiling tool still needs to print cuDF jar related information based on what version of RAPIDS+CUDF the Spark eventlog was based on. So I keep the example output with cuDF jar info there.

File: docs/FAQ.md
Comment (sameerz) [inline docs/FAQ.md:310]: Suggestion for this text and the Java API text below. 
```suggestion
Yes, but we don't actively test them, because the RAPIDS Accelerator hooks into Spark not at 
```
```suggestion
Yes, but we don't actively test them. It is because the RAPIDS Accelerator hooks into Spark not at 
```

**Review by sameerz — APPROVED** [review]

**Comment by sameerz** [comment]
build

**Comment by viadea** [comment]
Fixed a parameter typo in docs/additional-functionality/rapids-udfs.md:
spark.rapids.python.gpu.enabled -> spark.rapids.sql.python.gpu.enabled

@sameerz  would u mind re-approving?

**Review by sameerz — APPROVED** [review]

**Review by tgravescs — APPROVED** [review]

**Comment by tgravescs** [comment]
build

**Comment by tgravescs** [comment]
build

**Pull request merged**
